src/System.Text.Json/tests/Serialization/OptionsTests.cs

```diff
@@ -153,12 +153,18 @@ public static void ReadCommentHandling()
             int value = JsonSerializer.Parse<int>("1 /* commment */", options);
         }
 
-        [Fact]
-        public static void ReadCommentHandlingDoesNotSupportAllow()
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(JsonCommentHandling.Allow)]
+        [InlineData(3)]
+        [InlineData(byte.MaxValue)]
+        [InlineData(byte.MaxValue + 3)] // Other values, like byte.MaxValue + 1 overflows to 0 (i.e. JsonCommentHandling.Disallow), which is valid.
+        [InlineData(byte.MaxValue + 4)]
+        public static void ReadCommentHandlingDoesNotSupportAllow(int enumValue)
         {
             var options = new JsonSerializerOptions();
 
-            Assert.Throws<ArgumentException>(() => options.ReadCommentHandling = JsonCommentHandling.Allow);
+            Assert.Throws<ArgumentException>(() => options.ReadCommentHandling = (JsonCommentHandling)enumValue);
```

Comment:
Is there a name to validate in the exception?  Also, is this the right exception type?  We can't do anything stronger, like ArgumentOutOfRangeException?

Reply:
For JsonDocument.Parse it's ArgumentException because it's talking about the whole options payload.  For a property setter here, I agree that ArgumentOutOfRangeException is the way to go.